src/scsiencrypt.h

```diff
@@ -84,6 +86,46 @@ constexpr uint32_t BSLONG(const uint8_t *p)
 #define STENC_BIG_ENDIAN 0
 #endif
 
+// key-associated data
+struct __attribute__((packed)) scsi_kad {
+  uint8_t type;
```

Comment:
I recently learned that `std::uint8_t` is cleaner. Unfortunately I did not bookmark the page, where it was explained.

Reply:
Sure, it's more canonical C++ though a little verbose. I also switched to `std::byte` for bitfields.